### ISSUE TITLE
Not Parsing z-Coordinate

### ISSUE THREAD

**Issue by mjc-gh** [rgeo/activerecord-postgis-adapter#162]
Using Rails 4.2 with the latest, I have in my migration:
    create_table :items do |t|
      t.st_point :latlng, geographic: true, has_z: true
    end

In my model:

```
class Item < ActiveRecord::Base
  self.rgeo_factory_generator = RGeo::Geographic.spherical_factory({
    srid: 4326, has_z_coordinate: true
  })
end
```

In rails c:

```
Item.new(latlng: "POINT (1.0 2.0 3.0)").latlng
 => nil 

Item.new(latlng: "POINT (1.0 2.0)").latlng
=> #<RGeo::Geographic::SphericalPointImpl:0x4b9ba0 "POINT (1.0 2.0)"> 
```

When I then try to save it, PG complains about the missing Z-coordinate. Any ideas?

**Comment by ThomasPr** [comment]
Any updates on this? I ran into the same issue today. Using 3.0.0.beta2 with Rails 4.2.

**Comment by teeparham** [comment]
Currently the `rgeo_factory_generator` setting is ignored (see #163), which is the biggest issue holding up the 3.0 release.

However, in the specific case above, you should be able to remove the  `rgeo_factory_generator` call from your model, and it should create the correct factory, as of the latest master (see #176). I realize this isn't a long-term solution, but it should work for this specific problem.

**Comment by ThomasPr** [comment]
Thanks for getting back so quickly!

Unfortunately, this doesn't work either. Let me provide a very simple example:

```ruby
class CreateTours < ActiveRecord::Migration
  def change
    create_table :tours do |t|
      t.line_string :path, geographic: true, has_z: true
    end
  end
end
```

```ruby
class Tour < ActiveRecord::Base
end
```

I can run ```rake db:migrate``` without any problems and insert data using SQL:

```
                                   Table "public.tours"
 Column |            Type             |                     Modifiers                      
--------+-----------------------------+----------------------------------------------------
 id     | integer                     | not null default nextval('tours_id_seq'::regclass)
 path   | geography(LineStringZ,4326) | 
Indexes:
    "tours_pkey" PRIMARY KEY, btree (id)
```

```sql
# INSERT INTO tours (path) VALUES ('LINESTRING(1 1 1, 2 2 2)');
INSERT 0 1
```

```
# select * from tours;
 id |                                                            path                                                            
----+----------------------------------------------------------------------------------------------------------------------------
  1 | 01020000A0E610000002000000000000000000F03F000000000000F03F000000000000F03F000000000000004000000000000000400000000000000040
(1 row)
```

But I'm unable to access this data using ```rails console```

```ruby
>> Tour.last.path 
=> nil
```

**Comment by bbradbury** [comment]
Does running on https://github.com/bbradbury/activerecord-postgis-adapter/tree/feature/smart_column_factories yield any better results @ThomasPr ? Thinking https://github.com/rgeo/activerecord-postgis-adapter/pull/170/files#diff-0e74a421d0cce9a761df8f40cf1bdd42L54 might matter here

**Comment by ThomasPr** [comment]
@bbradbury Your branch works fine:

```ruby
>> Tour.last.path 
=> #<RGeo::Geographic::SphericalLineStringImpl:0x3fd302d88618 "LINESTRING (1.0 1.0 1.0, 2.0 2.0 2.0)">
```

**Comment by bbradbury** [comment]
@teeparham I added a few more comments to #170 :)

**Comment by teeparham** [comment]
@mikeycgto can you try this with the latest gem, version `3.0.0beta5`? You may need to set the factory using `SpatialFactoryStore` (see readme) if the default type isn't correct for linestring. It'd also be great to add a test for the scenario you mentioned.

**Comment by eliasbaixas** [comment]
maybe related to #280 ?

**Comment by teeparham** [comment]
I think this is fixed by #280
